cmake/third_party_deps.cmake

```diff
@@ -1,99 +1,111 @@
 include(FetchContent)
 
-# include pico-sdk
 macro(add_pico_sdk tag)
-  set(PICO_SDK_FETCH_FROM_GIT on)
-  set(PICO_SDK_FETCH_FROM_GIT_TAG ${tag})
-  set(PICOTOOL_FETCH_FROM_GIT_PATH ${CMAKE_SOURCE_DIR}/picotool)
-  include(${CMAKE_SOURCE_DIR}/cmake/pico_sdk_import.cmake)
+    set(PICO_SDK_FETCH_FROM_GIT on)
+    set(PICO_SDK_FETCH_FROM_GIT_TAG ${tag})
+    set(PICOTOOL_FETCH_FROM_GIT_PATH ${CMAKE_SOURCE_DIR}/picotool)
+    include(pico_sdk_import)
 endmacro()
 
-function(add_cexception)
-  FetchContent_Declare(
-    cexception
-    GIT_REPOSITORY https://github.com/ThrowTheSwitch/CException.git
-    GIT_TAG v1.3.3)
-  FetchContent_MakeAvailable(cexception)
-  add_library(CException__impl ${cexception_SOURCE_DIR}/lib/CException.c)
-  add_library(CException__hdrs INTERFACE)
-  target_include_directories(CException__hdrs
-                             INTERFACE ${cexception_SOURCE_DIR}/lib/)
-  add_library(CException INTERFACE)
-  target_link_libraries(CException INTERFACE CException__hdrs CException__impl)
-endfunction()
-
-function(add_unity)
-  FetchContent_Declare(
-    unity
-    GIT_REPOSITORY https://github.com/ThrowTheSwitch/Unity.git
-    GIT_TAG v2.5.2
-    OVERRIDE_FIND_PACKAGE)
-  FetchContent_MakeAvailable(unity)
-  find_package(unity)
-endfunction()
-
-function(add_runtime_c)
-  FetchContent_Declare(
-    elastic_ai_runtime_c
-    GIT_REPOSITORY https://github.com/es-ude/elastic-ai.runtime.c.git
-    GIT_TAG v2.7.2)
-  FetchContent_MakeAvailable(elastic_ai_runtime_c)
-  if(NOT topicMatcher__hdrs)
-    add_library(topicMatcher__hdrs INTERFACE)
+macro(init_pico_sdk)
+    pico_sdk_init()
+
+    include(pico_targets)
+    include(custom_targets)
+endmacro()
+
+macro(add_ctest)
+    include(CTest)
+endmacro()
+
+macro(add_cexception)
```

Comment:
I think the commit msg should explain why this was turned into a macro, because for the unknowing the change seems somewhat arbitrary.